prow/cmd/deck/job_history.go

```diff
@@ -0,0 +1,291 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/config"
+)
+
+const (
+	resultsPerPage = 20
+	idParam        = "buildId"
+
+	// ** Job history assumes the GCS layout specified here:
+	// https://github.com/kubernetes/test-infra/tree/master/gubernator#gcs-bucket-layout
+	logsPrefix    = "logs"
+	symLinkPrefix = "pr-logs/directory"
+)
+
+type BuildData struct {
+	index        int
+	SpyglassLink string
+	ID           string
+	Started      time.Time
+	Duration     time.Duration
+	Result       string
+}
+
+type JobHistoryTemplate struct {
+	OlderLink  string
+	NewerLink  string
+	LatestLink string
+	Name       string
+	Builds     []BuildData
+}
+
+type started struct {
+	Timestamp int64 `json:"timestamp"`
+}
+
+type finished struct {
```

Comment:
The started and finished structs are defined elsewhere in Prow. Should we try to maintain just one version of the structs or does it make more sense to specify the struct everywhere it is needed so we can just include the fields we want for that use case?

Reply:
AFAICT so far they're defined [here](https://github.com/kubernetes/test-infra/blob/master/prow/spyglass/viewers/metadata/metadata_viewer.go#L42). If we're planning to use a single version, I would probably move them to a more "central" location.